### ISSUE TITLE
[Build] No C++ library is generated after compilation completed

### ISSUE THREAD

**Issue by LiuPeiqiCN** [microsoft/onnxruntime#16610]
### Describe the issue

After compilation completed, I only found python whl and C# nuget package, but I canot found c++ library just like:
- install dir
-- bin
-- include
-- lib

### Target platform

Windows 11

### Build script

```.\build.bat --config Release --build_shared_lib --parallel --build --build_nuget --build_wheel --skip_tests --skip_onnx_tests --skip_winml_tests --skip_nodejs_tests --use_dnnl --use_dml --use_xnnpack --use_cuda --cuda_home "D:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.8" --cudnn_home "D:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.8" --cuda_version=11.8 --use_tensorrt --tensorrt_home "D:/vcpkgs/TensorRT/TensorRT-8.6.1.6" --use_openvino HETERO:GPU,CPU --use_azure --cmake_extra_defines BUILD_TESTING=OFF CMAKE_INSTALL_PREFIX="install" InferenceEngine_DIR="D:\vcpkgs\openvino\windows_2023.0.1.11005\runtime\cmake" ngraph_DIR="D:\vcpkgs\openvino\windows_2023.0.1.11005\runtime\cmake" onnxruntime_BUILD_UNIT_TESTS=OFF --cmake_generator "Visual Studio 17 2022" --update```

### Error / output

-

### Visual Studio Version

VS2022

**Comment by mszhanyi** [comment]
could you check if the sln file is generated correctly in {root of source}\build\Windows ?

**Comment by LiuPeiqiCN** [comment]
Yes, sln file is generated correctly and I can build ONNXRuntime successful.
It seems that ONNXRuntime only generates the lib files in src\build\Windows\Release, but it does not perform the installation action for the package. If I manually open the sln file and execute the "Install" project, there will be a lot of missing files in the generated bin/include/lib folders.
Furthermore, even if I integrate CUDA/TensorRT/OpenVINO, the generated C# NuGet package only supports OpenVINO and lacks the relevant files for CUDA/TensorRT.

**Comment by snnn** [comment]
There is no formal installation step. ONNX Runtime is a library. To use it, you need:
1. onnxruntime_c_api.h and its dependencies 
2. The DLLs generated from your build, mainly onnxruntime.dll
3. The import libs of the DLLs

You can copy out the files to somewhere then start to use them.  You can also get a packed zip file from https://github.com/microsoft/onnxruntime/releases, then unzip it you will know what files from your build are necessary.

**Comment by snnn** [comment]
Here is the script: https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/github/azure-pipelines/templates/c-api-artifacts-package-and-publish-steps-windows.yml
